include/boost/geometry/index/rtree.hpp

```diff
@@ -431,13 +426,156 @@ class rtree
                           allocator_type const& allocator = allocator_type())
         : m_members(getter, equal, parameters, allocator)
     {
-        typedef detail::rtree::pack<members_holder> pack;
-        size_type vc = 0, ll = 0;
-        m_members.root = pack::apply(::boost::begin(rng), ::boost::end(rng), vc, ll,
-                                     m_members.parameters(), m_members.translator(),
-                                     m_members.allocators());
-        m_members.values_count = vc;
-        m_members.leafs_level = ll;
+        pack_construct(::boost::begin(rng), ::boost::end(rng), boost::container::new_allocator<void>());
+    }
+
+    /*!
+    \brief The constructor.
+
+    The tree is created using packing algorithm and a temporary packing allocator.
+
+    \param first             The beginning of the range of Values.
+    \param last              The end of the range of Values.
+    \param parameters        The parameters object.
+    \param getter            The function object extracting Indexable from Value.
+    \param equal             The function object comparing Values.
+    \param tree_allocator    The allocator object for persistent data in the tree..
+    \param pack_allocator    The temporary allocator object used when packing
```

Comment:
I think for consistency with other constructors I'd prefer keeping the old name `allocator` instead of `tree_allocator`. I think it's clear enough.

What do you think about changing the name of `pack_allocator` to `temp_allocator`? I think it would be more clear what is happening there, i.e. it's an allocator used in construction of temporary data.

Reply:
I agree.